libfwupdplugin/fu-device-locker.c

```diff
@@ -189,6 +189,7 @@ fu_device_locker_new_full(gpointer device,
 				}
 			}
 		}
+		g_object_unref(device);
```

Comment:
this isn't correct; if open_func() fails then it **has** to return false, and `has` to set a GError.